### DISCUSSION TITLE
Pydantic v2

### DISCUSSION THREAD

**Discussion by tomvonheill** [aws-powertools/powertools-lambda-python#2882] [General]
Just wondering if there is a plan in place to migrate to pydantic v2? timelines/issues/etc all welcomed

Found this, but was wondering if there is a larger discussion being had?
I have seen the following https://github.com/aws-powertools/powertools-lambda-python/pull/2733

**Comment by leandrodamascena** [comment]
Hello @tomvonheill! Since Powertools v2.21.0 we already support Pydantic v2 with Powertools! We kept our default version of Pydantic as v1 (and same validators/functions) to avoid any breaking changes, but if you want you can use Pydantic v2 with Powertools.

I don't know if you expect anything different from this and it might be good to get your point of view.

PR: https://github.com/aws-powertools/powertools-lambda-python/pull/2733
Documentation: https://docs.powertools.aws.dev/lambda/python/2.22.0/utilities/parser/#using-pydantic-v2

**Reply by leandrodamascena** [reply]
Just adding another info: we expect to drop support for Pydantic v1 in Powertools V3, but we don't have an ETA yet.

**Reply by leandrodamascena** [reply]
Hello @tomvonheill! Do you still need some help?

Thank you.

**Comment by leandrodamascena** [comment]
Hey @tomvonheill! I'm closing this discussion as completed. 

Please re-open it if needed.
